### PR TITLE
Pype Blender Integration

### PR DESCRIPTION
**Pull request by simonebarbieri**
This PR implements basic support for Blender.

It supports 4 families:
- Model
- Rig
- Animation
- Action

While Model, Rig and Animation are similar to other platforms, the Action family is specific to Blender to save and load common animations for specific characters.

Models, rigs and animation can be published as .blend and/or .fbx, but can be linked from .blend files only.

### CODE REVIEW

File: pype/plugins/blender/publish/collect_animation.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    def get_animation_collections() -> Generator:
+        """Return all 'animation' collections.
+
+        Check if the family is 'animation' and if it doesn't have the
+        representation set. If the representation is set, it is a loaded
+        animation and we don't want to publish it.
+        """
+        for collection in bpy.data.collections:
+            avalon_prop = collection.get(AVALON_PROPERTY) or dict()
+            if (avalon_prop.get('family') == 'animation'
```
[the remark this answers is not in the document]
Comment (mkolar) [reply pype/plugins/blender/publish/collect_animation.py]: I've done this already.

File: pype/plugins/blender/load/load_animation.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+    families = ["animation"]
+    representations = ["blend"]
+
+    label = "Link Animation"
+    icon = "code-fork"
+    color = "orange"
+
+    @staticmethod
+    def _remove(self, objects, lib_container):
```
Comment (antirotor) [inline pype/plugins/blender/load/load_animation.py]: I am not sure why those methods in Loader plugin are market as static methods. I couldn't find place where they are statically called and when they are, `self` is passed manually.

File: pype/plugins/blender/load/load_rig.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+        rig_container = scene.collection.children[lib_container].make_local()
+
+        meshes = [obj for obj in rig_container.objects if obj.type == 'MESH']
+        armatures = [
+            obj for obj in rig_container.objects if obj.type == 'ARMATURE']
+
+        objects_list = []
+
+        assert(len(armatures) == 1)
```
Comment (antirotor) [inline pype/plugins/blender/load/load_rig.py]: why this assert? It should throw some error message and there should be corresponding validator if this is enforced

File: pype/plugins/blender/load/load_animation.py (hunk 2 above)
Comment (simonebarbieri) [reply pype/plugins/blender/load/load_animation.py]: You are right, I'm not sure why I set them as static! I'll change that.

File: pype/plugins/blender/load/load_rig.py (hunk 3 above)
Comment (mkolar) [reply pype/plugins/blender/load/load_rig.py]: we'll make a ticket for this. but merging for now, as this is good enough for testing, to make on it's way to production.

**Pull request merged**